support-frontend/assets/components/ticker/contributionTicker.jsx

```diff
@@ -61,56 +68,77 @@ export default class ContributionTicker extends Component<PropTypes, StateTypes>
-      }, 500);
+      window.requestAnimationFrame(this.increaseTextCounter);
     });
   }
 
   onGoalReached: () => void;
   tickerJsonUrl: string;
   totalSoFar: number;
   count = 0;
-  filledProgressBar: ?HTMLDivElement;
-  increaseCounter = this.increaseCounter.bind(this);
-
-  animateBar(totalSoFar: number, goal: number) {
-    const progressBarElement = this.filledProgressBar;
-    if (progressBarElement && progressBarElement instanceof HTMLElement) {
-      progressBarElement.style.transform = `translateX(${percentageTotalAsNegative(totalSoFar, goal)}%)`;
-    }
-  }
+  tickerType: TickerType;
+  classModifiers: Array<?string>;
+  goalReached: boolean;
 
-  increaseCounter() {
+  increaseTextCounter = () => {
     const { totalSoFar } = this;
     this.count += Math.floor(totalSoFar / 100);
 
     if (this.count >= this.totalSoFar) {
       this.setState({ totalSoFar: this.totalSoFar });
     } else {
       this.setState({ totalSoFar: this.count });
-      window.requestAnimationFrame(this.increaseCounter);
+      window.requestAnimationFrame(this.increaseTextCounter);
     }
   }
 
+  renderContributedSoFar = () => {
+    if (!this.goalReached) {
+      return (
+        <div className="contributions-landing-ticker__so-far">
+          <div className="contributions-landing-ticker__count">${Math.floor(this.state.totalSoFar).toLocaleString()}</div>
+          <div className="contributions-landing-ticker__count-label contributions-landing-ticker__label">contributed</div>
+        </div>
+      );
+    }
+
+    if (this.goalReached && this.tickerType === 'unlimited') {
+      return (
+        <div className="contributions-landing-ticker__so-far">
+          <div className="contributions-landing-ticker__count">We&#39;ve met our goal &mdash; thank you</div>
+          <div className="contributions-landing-ticker__count-label contributions-landing-ticker__label">Contributions are still being accepted</div>
+        </div>
+      );
+    }
+
+    if (this.goalReached && this.tickerType === 'hardstop') {
+      return (
+        <div className="contributions-landing-ticker__so-far">
+          <div className="contributions-landing-ticker__count">We&#39;ve met our goal &mdash; thank you</div>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
 
   render() {
+    const readyToRender = (this.state && this.state.totalSoFar && this.state.totalSoFar !== 0);
+    const allClassModifiers = readyToRender ? this.classModifiers : [...this.classModifiers, 'hidden'];
     const baseClassName = 'contributions-landing-ticker';
-    const wrapperClassName = (this.state && this.state.totalSoFar && this.state.totalSoFar !== 0) ?
-      baseClassName :
-      classNameWithModifiers(baseClassName, ['hidden']);
+    const wrapperClassName = classNameWithModifiers(baseClassName, allClassModifiers);
+    const progressBarAnimation = `translate3d(${percentageToTranslate(this.state.totalSoFar, this.state.goal, this.tickerType)}%, 0, 0)`;
```

Comment:
can we have a comment explaining why we're doing it with a `translate3d` (it's GPU acceleration right?)

Reply:
yeah any of the properties that act on the `z` access give you gpu acceleration